### ISSUE TITLE
"No files are available to open" when selecting files in container

### ISSUE THREAD

**Issue by SetTrend** [microsoft/vscode-docker#2983]
I can hardly open files in a Docker container. Most of the time, when I do, I get an error message telling me that "no files are avialable to open":

![Files won't open](https://user-images.githubusercontent.com/9283914/120624989-865c4d80-c461-11eb-9ec4-b4d2b30381a5.gif)

I propose that when the VS Code Docker extension is in a state where it cannot retrieve container files, the corresponding Tree control should appear disabled until files are ready to be transferred again.

**Comment by philliphoff** [comment]
@SetTrend I don't see that same behavior.  What I *do* see in the video is the Containers tree refreshing an *awful* lot, which could be a problem (e.g. the tree clears the selected node out from under the command trying to use it to know what to open).  That specific error message comes from logic that's managing multi-select, which would make me suspect it's a refresh issue.

How have you setup Docker on your machine?  What platform are you using, version numbers, etc.?  Is the extension connected to the local Docker daemon, or perhaps a remote one (which might have more latency which isn't being handled well by the refresh logic)?  Do you have a custom refresh rate set (under the Docker settings)?

**Comment by lruffing** [comment]
I see the same issue. And the refreshing does occur a lot. I don't believe I have any special settings on the Docker extension.
I see there is this one setting but it appears to be the default value. Can you verify that?
![image](https://user-images.githubusercontent.com/8334304/120716978-a993f680-c47b-11eb-8b7a-102da5090a2e.png)
I am using Docker for Windows and I usually have Docker Desktop running. I'm wondering if you do as well @SetTrend ?

**Comment by SetTrend** [comment]
I'm using Docker for Windows. Here's the version information (should be "latest"):

```
PS > docker version
Client:
 Cloud integration: 1.0.14
 Version:           20.10.6
 API version:       1.41
 Go version:        go1.16.3
 Git commit:        370c289
 Built:             Fri Apr  9 22:49:36 2021
 OS/Arch:           windows/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.6
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       8728dd2
  Built:            Fri Apr  9 22:44:56 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.4
  GitCommit:        05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc:
  Version:          1.0.0-rc93
  GitCommit:        12644e614e25b05da6fd08a38ffa0cfe1903fdec
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```
<br/>

I didn't configure any Docker daemon settings. I'm using the default configuration Docker comes with.

![Docker settings](https://user-images.githubusercontent.com/9283914/120721555-5eef9a00-c4ce-11eb-9b02-311c2674ee19.png)
<br/>

The VM I'm accessing is a Linux container [provided by Oracle](/oracle/docker-images).

Does this information help you?

**Comment by bwateratmsft** [comment]
I'm able to reproduce this immediately. I think it is indeed related to the over-refreshing that's going on.

**Comment by bwateratmsft** [comment]
It seems like this maybe went away on its own in the latest build from main, possibly due to the new `vscode-azureextensionui` in #3077? I'd like to investigate and get positive confirmation.

**Comment by bwateratmsft** [comment]
Ok, I think this was regressed by #2733 and/or #2746. During a container's first minute of life, it's constantly updating the timestamp.

Later on, it starts updating status only every minute, and probably after its first hour, the update cadence becomes hours. Needless to say, the issue is far, far more noticeable during that first minute.

**Comment by bwateratmsft** [comment]
A fix for this has been released in Docker extension version 1.16.0.

**Comment by SetTrend** [comment]
Excellent! Thank you for keeping up the good work! 👍👍👍
